packages/flame/lib/src/sprite_animation.dart

```diff
@@ -308,6 +318,10 @@ class SpriteAnimation {
   bool _done = false;
   bool done() => _done;
 
+  /// Local flag to determine if the animation has started to prevent multiple
+  /// calls to [onStart].
+  bool _started = false;
```

Comment:
This variable may not be needed if you let `currentIndex==-1` or `currentIndex==null` signify that the animation hasn't started yet.

Reply:
Looking at this, what is the impact if its set -1?  We would need to increment it first in the !_started if block, right?  Also, now isFirstFrame is no longer being used, so we can drop that as well.

Reply:
I think you can leave it, it feels like a premature optimization to remove that bool (but it would be possible with both starting at a negative index and having it nullable).

Reply:
Ok.  I think for now, it's all working and I think I would like to leave it as is. I can think on it a bit more, but for now I'm going to work on something else, if ok.